pkg/virt-handler/vm.go

```diff
@@ -441,6 +442,28 @@ func (d *VirtualMachineController) execute(key string) error {
 	return nil
 }
 
+func (d *VirtualMachineController) replacePVCByHostDisk(vmi *v1.VirtualMachineInstance) error {
+	// if PVC is defined then it is replaced by HostDisk
+	// PersistenVolumeClaim is mounted into pod as directory from node filesystem
+	// it is not attached as independent block device
+	for i := range vmi.Spec.Volumes {
+		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.PersistentVolumeClaim != nil {
+			pvcSize, err := hostdisk.GetPVCSize(volumeSource.PersistentVolumeClaim.ClaimName, vmi.Namespace, d.clientset)
+			if err != nil {
+				return err
+			}
+			volumeSource.HostDisk = &v1.HostDisk{
+				Path:     hostdisk.GetPVCDiskImgPath(vmi.Spec.Volumes[i].Name),
+				Type:     v1.HostDiskExistsOrCreate,
+				Capacity: pvcSize,
+			}
+			// PersistenVolumeClaim is replaced by HostDisk
+			volumeSource.PersistentVolumeClaim = nil
+		}
+	}
+	return nil
+}
+
 func (d *VirtualMachineController) injectCloudInitSecrets(vmi *v1.VirtualMachineInstance) error {
```

Comment:
Either we rename this method or we should create a separate method for injecting the additional PVC data.

Reply:
Forget the comment.